tests/purs/failing/ConstraintInference.out

```diff
@@ -4,7 +4,7 @@ at tests/purs/failing/ConstraintInference.purs:10:1 - 10:21 (line 10, column 1 -
 
   The inferred type
   [33m                                         [0m
-  [33m  forall t8 t11. Show t8 => t11 -> String[0m
+  [33m  forall c8 t11. Show c8 => t11 -> String[0m
```

Comment:
Due to my hacky implementation of this error message, line 11 of the output which reads `t8 could not be determined` is broken, since it should read `c8` now. Is there any easy way to do the right thing when printing error messages, or should it be accessed when generating the error message (e.g. passing `[Text]` instead of `[Int]` in the error constructor `AmbiguousTypeVariables SourceType [Int]`)?

Here are the relevant changes from my PR, lmk if you have questions: https://github.com/purescript/purescript/pull/3721/files#diff-11e3ee9ff014e25b38a3dfdaa1fa18c8d0ca9dab1860d355abad0034978ea04cL691-R700

I will try to look at it myself soon but I can't make any promises.

Reply:
I feel like changing the error constructor to `[Text]` would be simpler here, since the information comes from `CheckState`

Reply:
I used a tuple `[(Text, Int)]` as to preserve information